homeassistant/components/group.py

```diff
@@ -161,48 +169,82 @@ def reload_service_handler(service_call):
     return True
 
 
-def _process_config(hass, config, component):
+@asyncio.coroutine
+def _async_process_config(hass, config, component):
     """Process group configuration."""
+    groups = []
     for object_id, conf in config.get(DOMAIN, {}).items():
         name = conf.get(CONF_NAME, object_id)
         entity_ids = conf.get(CONF_ENTITIES) or []
         icon = conf.get(CONF_ICON)
         view = conf.get(CONF_VIEW)
 
-        group = Group(hass, name, entity_ids, icon=icon, view=view,
-                      object_id=object_id)
-        component.add_entities((group,))
+        # This order is important as groups get a number based on creation
+        # order.
```

Comment:
I think that is not a problem. But we can add a boolean running and set it true and at the end false. We ned only check if it true and return the function if a process is allready in progress.